commands.go

```diff
@@ -429,6 +429,29 @@ func (c *Client) Set(key, value string) *StatusCmd {
 	return cmd
 }
 
+type SetOpt struct {
+	EX, PX time.Duration
+	NX, XX bool
+}
+
+func (c *Client) SetSpecial(key, value string, opt SetOpt) *StatusCmd {
```

Comment:
I would name it `SetExt` or `SetExtended` if we are going to add it into v2 branch.